utils/check/handle_rpms.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+set -uex
+
+# This file does an install of the RPMs found in "${REPO_DIR}" and then
+# finds all the packages that the binaries installed reference.
+
+# The file ${PACKAGES} will contain a list of packages that the RPMs depend
+# on.
+
+# The file ${SOFILES} will contain a list of sharable images depended on by
+# the RPMs.  The default name for ${SOFILES} is "daos_depends_libries".
+
+# This will likely need to be edited for other build systems.
+
+: "${WORKSPACE:="${PWD}"}"
+
+: "${RPM_DIR:="${WORKSPACE}/rpm_dir"}"
+
+: "${SOFILES:="daos_depends_libraries"}"
+: "${PACKAGES:="daos_depends_packages"}"
+
+
+repo_file="/etc/yum.repos.d/"
+repo_file+="repo.dc.hpdd.intel.com_repository_daos-stack-"
+repo_file+="el-7-x86_64-stable-local.repo"
+
+repo_url="https://repo.dc.hpdd.intel.com/repository/"
+repo_url+="daos-stack-el-7-x86_64-stable-local"
+
+if [ ! -e "${repo_file}" ]; then
+  cat << EOF > daos_repo_file
+
+[repo.dc.hpdd.intel.com_repository_daos-stack-el-7-x86_64-stable-local]
+name=added from: ${repo_url}
+baseurl=${repo_url}
+enabled=1
+
+gpgcheck = False
+
+EOF
+  sudo cp daos_repo_file "${repo_file}"
+  sudo yum -y makecache
+fi
+
+sudo yum -y remove cart cart-debuginfo cart-devel cart-tests \
+                   daos daos-client daos-debuginfo daos-devel daos-server \
+                   daos-tests CUnit dpdk fio fuse fuse-lib hwloc \
+                   libabt0 libcmocka libfabric libfabric-devel libisa-l \
+                   libopa1 libpmem libpmemblk mercury ndctl\
+                   ompi openmpi3 openpa-devel \
+                   pexpect protobuf-c pmix \
+                   python-configshell python-urwid \
+                   raft raft-debuginfo raft-devel \
+                   spdk spdk-tools
```

Comment:
This seems very brittle to the ongoing changes in dependencies of DAOS.  This is going to get forgotten and bit-rotten very quickly.  It's already out of date in fact as raft{,-devel} are no longer installed with DAOS.

Reply:
In general this needs to be a list of packages that were built by us that may be installed on the test node from a previous test run.

And this procedure may need to be run on multiple release branches or master for something we are building.

This procedure generally only needs to be run once per release on a test VM.  If components are not present the procedure does not break.   If there are more components to be removed, then the person running the test will need to edit the script and submit a PR with the updated script to record the results.

It is a bit more work that I want to do at this time to make any more generic and a start at that would be:
For each rpm to be installed do a:
yum -q deplist $rpm | grep "provider" | sort -u >> list
Convert the list to remove the "provider: * and sort, etc, to make the list just contain packages to remove.
Remove from that list all the packages that are normally installed as part of the base OS and should not be removed such as bash, coreutils.  I am sure that there is a list of these somewhere or a way to get them.  I do not plan on taking the time to look for how to do that.